File: app/src/main/java/com/aus/ausgegeben/data/DataSeeder.kt
package com.aus.ausgegeben.data

import com.aus.ausgegeben.data.entity.Category
import com.aus.ausgegeben.ui.TransactionType
import kotlinx.coroutines.flow.first

object DataSeeder {
    suspend fun seedIfEmpty(repository: AppRepository) {
        val categories = repository.allCategories.first()
        if (categories.isNotEmpty()) return

        val expense = TransactionType.EXPENSE.storageKey
        val income = TransactionType.INCOME.storageKey
        val transfer = TransactionType.TRANSFER.storageKey

        fun cat(name: String, icon: String, color: Int, type: String, order: Int) =
            Category(
                name = name,
                iconName = icon,
                colorInt = color,
                transactionType = type,
                sortOrder = order
            )

        repository.insertAllCategories(
            listOf(
                cat("Groceries", "shopping_cart", 0xFFE86B5A.toInt(), expense, 0),
                cat("Shopping", "shopping_bag", 0xFFE8A060.toInt(), expense, 1),
                cat("Dining", "restaurant", 0xFFD4849A.toInt(), expense, 2),
                cat("Transport", "car", 0xFF6A9FD4.toInt(), expense, 3),
                cat("Bills", "bolt", 0xFF9A8FD4.toInt(), expense, 4),
                cat("Subscriptions", "subscriptions", 0xFF5AB8AA.toInt(), expense, 5),
                cat("Salary", "credit_card", 0xFF5CB88A.toInt(), income, 0),
                cat("Freelance", "work", 0xFF6A9FD4.toInt(), income, 1),
                cat("Refunds", "undo", 0xFFB8A060.toInt(), income, 2),
                cat("Transfer", "swap_horiz", 0xFF8E8E96.toInt(), transfer, 0)
            )
        )
    }
}

File: web/src/services/dataSeeder.ts
import type { Category } from '@/models/types';

export const seedCategories: Omit<Category, 'id'>[] = [
  { cloudId: '1f2e1125-2a9f-42bd-8747-8901c7d14016', name: 'Groceries', iconName: 'shopping_cart', colorInt: 0xffe86b5a, transactionType: 'expense', sortOrder: 0 },
  { cloudId: '9bc5ceb6-9074-429b-a51e-575161edf1e3', name: 'Shopping', iconName: 'shopping_bag', colorInt: 0xffe8a060, transactionType: 'expense', sortOrder: 1 },
  { cloudId: '0d60f4c0-ee4e-43e1-986b-3556f9a5bec7', name: 'Dining', iconName: 'restaurant', colorInt: 0xffd4849a, transactionType: 'expense', sortOrder: 2 },
  { cloudId: 'fda6a539-33ba-4982-86cc-5893eb8c08dc', name: 'Transport', iconName: 'car', colorInt: 0xff6a9fd4, transactionType: 'expense', sortOrder: 3 },
  { cloudId: '00062709-f071-43b9-a23c-c6487e456f97', name: 'Bills', iconName: 'bolt', colorInt: 0xff9a8fd4, transactionType: 'expense', sortOrder: 4 },
  { cloudId: '8dfd6f47-76bb-4f66-979d-e2d82d49c9a7', name: 'Subscriptions', iconName: 'subscriptions', colorInt: 0xff5ab8aa, transactionType: 'expense', sortOrder: 5 },
  { cloudId: '62151dd1-e910-4714-a1be-cdf7be95beb7', name: 'Salary', iconName: 'credit_card', colorInt: 0xff5cb88a, transactionType: 'income', sortOrder: 0 },
  { cloudId: 'ff830f1d-cf14-4f95-b0d7-dc9f160fed51', name: 'Freelance', iconName: 'work', colorInt: 0xff6a9fd4, transactionType: 'income', sortOrder: 1 },
  { cloudId: '0ebe5034-1c55-48e4-84c0-721809530136', name: 'Refunds', iconName: 'undo', colorInt: 0xffb8a060, transactionType: 'income', sortOrder: 2 },
  { cloudId: '9a251df8-6d84-4603-a53b-ffe470dfa433', name: 'Transfer', iconName: 'swap_horiz', colorInt: 0xff8e8e96, transactionType: 'transfer', sortOrder: 0 },
];
